lib/ably/rest/client.rb

```diff
@@ -434,6 +439,16 @@ def send_request(method, path, params, options)
         max_retry_duration = http_defaults.fetch(:max_retry_duration)
         requested_at       = Time.now
         retry_count        = 0
+        if @add_request_ids
+          if method == :get
+            path_with_params = Addressable::URI.new
+          else
+            path_with_params = Addressable::URI.parse(path)
+          end
+          path_with_params.query_values = {request_id: SecureRandom.hex(20)}
+          query = path_with_params.query
+          path = "#{path}?#{query}"
+        end
```

Comment:
Why are you manually munging the path? This method takes a params hash. You can add it to the params hash.

Reply:
@SimonRichardson because this is Ruby and not Elixir.  So the `params` object would be changed by just calling this method, which is bad.

Reply:
> So the params object would be changed by just calling this method, which is bad.

@mattheworiordan fair enough, but imo the solution to that is to clone the hash (as Cesare has done in the latest version of this pr), not to start munging a param into the path (meaning faraday has to parse it out again to combine with the params -- assuming it even does that rather than give you a path with two `?`s). (for one thing the original approach assumes the path doesn't already have a querysting, so fails the 'what if two things did this' test)